frontend/src/components/P5Wrapper.jsx

```diff
@@ -16,23 +17,38 @@ const P5Wrapper = () => {
   useEffect(() => {
     // Initialize p5.js instance and attach it to the sketchRef DOM node
     let sketch;
-    switch (gameContext.gamemodeType){
-      case "Circlefall":
-        sketch = Circlefall;
-        break;
-      case "Gridshot":
-        sketch = Gridshot;
-        break
-      default:
-        sketch = Circlefall;
-    }
-    const p5Instance = new p5((p) => sketch(p, gameContext.gamemodeDataFilePath, gameContext.dispatch), sketchRef.current);
+    let gamemodeDataFilePath;
+
+    const paths = {
+      Circlefall: {
+          Normal: { sketch: Circlefall, filePath: "CirclefallNormal.json" },
+          Hard: { sketch: Circlefall, filePath: "CirclefallHard.json" },
+          Impossible: { sketch: Circlefall, filePath: "CirclefallImpossible.json" },
+          Marathon: { sketch: CirclefallWave, filePath: "CirclefallMarathon.json" },
+          default: { sketch: Circlefall, filePath: "CirclefallNormal.json" }
+      },
+      Gridshot: {
+          Classic: { sketch: Gridshot, filePath: "GridshotClassic.json" },
+          Mini: { sketch: Gridshot, filePath: "GridshotMini.json" },
+          default: { sketch: Gridshot, filePath: "GridshotClassic.json" }
+      },
+      default: { sketch: Circlefall, filePath: "CirclefallNormal.json" }
+    };
+
+    const mode = paths[gameContext.gamemode.mode] || paths.default;
+    const type = mode[gameContext.gamemode.type] || mode.default;
+
+    sketch = type.sketch;
+    gamemodeDataFilePath = type.filePath;
+    console.log(gamemodeDataFilePath);
```

Comment:
This was probably left here on accident, I just wanted to bring attention to it just in case you did want to remove it.

Reply:
Good catch, thanks